src/ArrowFunctions.cpp

```diff
@@ -128,16 +128,9 @@ int cpp_readColumnByName(const char* filename, void* chpl_arr, const char* colna
       
       column_reader = row_group_reader->Column(idx);
 
-      if(ty == ARROWINT64) {
-        auto chpl_ptr = (int64_t*)chpl_arr;
-        parquet::Int64Reader* reader =
-          static_cast<parquet::Int64Reader*>(column_reader.get());
-
-        while (reader->HasNext()) {
-          (void)reader->ReadBatch(batchSize, nullptr, nullptr, &chpl_ptr[i], &values_read);
-          i+=values_read;
-        }
-      } else if(ty == ARROWUINT64) {
+      // Since int64 and uint64 Arrow dtypes share a physical type and only differ
+      // in logical type, they must be read from the file in the same way
+      if(ty == ARROWINT64 || ty == ARROWUINT64) {
```

Comment:
This seems surprising. At the very least I would expect you need to cast the chapel array to `uint64_t*` and I'm surprised the parquet types aren't different. Can you add a test that verifies reading in `max(uint)` works correctly?

And anything needed for reading in uint32?

Reply:
> I would expect you need to cast the chapel array to uint64_t*

That was my initial thought as well, but Parquet does not even expose anything to read in `uint` values, so that does not work. They are both 64 bit and share the same physical type, so I think that makes sense.

> Can you add a test that verifies reading in max(uint) works correctly?

Added.

> And anything needed for reading in uint32?

I hadn't considered this since I just had `uint64` support in my head, but I've added it now. Since we don't support writing uint32 (or int32), we won't be testing it, but I saw it work locally with some struggles to create the file in pyarrow.